src/transaction/algo.ts

```diff
@@ -0,0 +1,54 @@
+const algosdk = require('algosdk');
+const base32 = require('base32.js');
+
+export const getClient = (testnet: boolean) => {
+    if (testnet) {
+        const baseServer = 'https://testnet-algorand.api.purestake.io/ps2'
+        const port = '';
+        const token = {'X-API-Key': 'klnvbVsn983Q3awiuWUGn2OKHkyi3U974JYtGvdt'}
+        const algodClient = new algosdk.Algodv2(token, baseServer, port);
+        return algodClient;
+    } else {
+        return null;
+    }
+}
+
+const waitForConfirmation = async (algodClient: any, txId: string) => {
+    let lastround = (await algodClient.status().do())['last-round'];
+    let limit = 0;
+    while (limit < 30) {
+        let pendingInfo = await algodClient.pendingTransactionInformation(txId).do();
+        if (pendingInfo['confirmed-round'] !== null && pendingInfo['confirmed-round'] > 0) {
+            return 1;
+            break;
+        }
+        lastround++;
+        limit++;
+        await algodClient.statusAfterBlock(lastround).do();
+    }
+    return 0;
+}
+
+export const signAlgoTransaction = async ( testnet: boolean, from: string, to: string, fee: number, amount: BigInt, privKey: string) => {
+    let algodClient = getClient(testnet);
```

Comment:
make the parameter 1 Class - AlgoTransaction.ts in the /request/model section with validations on the fields. amount should be string with internal conversion. use fromPrivateKey for PK, also need to use PrivateKeyOrSignatureId as a base class for that - for transactions, which should be signed via Tatum KMS, where private key is not available